pig/src/main/java/org/jboss/pnc/bacon/pig/PigFacade.java

```diff
@@ -161,29 +165,45 @@ public static GroupBuildInfo run(
         // TODO: should we bother with this case?
 
         RepositoryData repo = null;
-
-        if (repoZipPath != null) {
-            repo = parseRepository(new File(repoZipPath));
+        
+        if (repoZipPath != null || context.getPigConfiguration()
+                .getFlow()
+                .getRepositoryGeneration()
+                .getStrategy() != RepoGenerationStrategy.IGNORE) {
+            if (repoZipPath != null) {
+                repo = parseRepository(new File(repoZipPath));
+            } else {
+                repo = generateRepo(removeGeneratedM2Dups, configurationDirectory, strictLicenseCheck);
+            }
+            context.setRepositoryData(repo);
+            context.storeContext();
         } else {
-            repo = generateRepo(removeGeneratedM2Dups, configurationDirectory, strictLicenseCheck);
+            log.info("Skipping Repo Generation");
         }
-        context.setRepositoryData(repo);
-        context.storeContext();
 
-        if (!skipSources) {
+        if (!(skipSources || context().getPigConfiguration()
```

Comment:
if skipSources is true or strategy is ignore we're skipping

Reply:
Should that be `!= IGNORE` ?